internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -561,6 +593,27 @@ func (c *ClusterManager) reconcileClusterDNS(cluster api.Cluster) error {
 	return nil
 }
 
+func (c *ClusterManager) reconcileKasFleetshardOperator(cluster api.Cluster) error {
+	if c.KasFleetshardOperatorAddon != nil {
+		if params, err := c.KasFleetshardOperatorAddon.ReconcileParameters(cluster); err != nil {
+			if err.IsBadRequest() {
+				glog.Infof("kas-fleetshard operator is not found on %s cluster %s", cluster.Status, cluster.ClusterID)
```

Comment:
Question: bad request error is returned when operator is not found instead of `not found`?

Reply:
Hi @ziccardi thanks for the feedback. This was pre-existing code, so I am unfamiliar with the error which should be returned in this case.
___
`IsBadRequest()` - BadRequest() occurs when the request is unable to be procesed
___

`NotFound()` - NotFound occurs when a record is not found in the database
___

To clarify, would you be suggesting to return `NotFound`, or return `BadRequest()` and replace the error message?

Reply:
I'm just confused because the code checks if the error is a `bad request` and, if it is, logs a `not found`.
So my comment was about checking that `ReconcileParameters` is really returning a `BadRequest` error when the operator is not found.

Reply:
Ok, thanks @ziccardi. Would you have a recommendation for how I should check this?

Reply:
@VanillaSpoon I tried to give a quick look at [ReconcileParameters](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L97).

It explicitly returns an `ErrorGeneral` a number of times.

The only other error that is returned is `paramErr` [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L100)

So, looking inside [o.getAddonParams](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L125) I see it is returning again a `GeneralError`.

Unless I miss something, I would conclude that `BadRequest` is never returned from `ReconcileParameters`

Reply:
Thank you @ziccardi

Reply:
> @VanillaSpoon I tried to give a quick look at [ReconcileParameters](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L97).
> 
> It explicitly returns an `ErrorGeneral` a number of times.
> 
> The only other error that is returned is `paramErr` [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L100)
> 
> So, looking inside [o.getAddonParams](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5ca5ce3f8bf99d88c53c2e62b1e1ec6220be3f37/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L125) I see it is returning again a `GeneralError`.
> 
> Unless I miss something, I would conclude that `BadRequest` is never returned from `ReconcileParameters`

That's correct. Thanks for the investigation @ziccardi. 
Based on it, would be able to suggest a way forward? I can think of removing this https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/3ed563d459a8796c2681771fbacca9f69629df61/internal/kafka/internal/workers/clusters_mgr.go#L594-L596 condition. Would you agree @ziccardi @VanillaSpoon ?

Reply:
Thanks @VanillaSpoon. Let's also get @ziccardi opinion on it before going forward with the code change.

Reply:
Agree on removing that. It is useless and creates confusion